packages/atomic/src/components/facets/facet-search/facet-search.tsx

```diff
@@ -1,52 +1,199 @@
+      <button
+        type="button"
+        part="clear-button"
+        class="clear-button mr-2"
+        onClick={() => {
+          this.props.controller.text = '';
+          this.inputRef.focus();
+        }}
+      >
+        <div
+          innerHTML={ClearIcon}
+          class="text-on-background fill-current h-2.5 w-2.5"
+        />
+      </button>
+    );
+  }
+
+  private get input() {
     return (
       <input
-        onInput={(e) =>
-          this.facetSearch.emit((e.target as HTMLInputElement).value)
-        }
-        class="apply-border-on-background-variant w-full rounded"
+        part="search-input"
+        ref={(el) => (this.inputRef = el as HTMLInputElement)}
+        onFocus={() => {
+          this.props.controller.state.showFacetSearchResults = true;
+        }}
+        onBlur={() => this.combobox.onInputBlur()}
+        onInput={(e) => this.combobox.onInputChange(e)}
+        onKeyUp={(e) => this.combobox.onInputKeyup(e)}
+        onKeyDown={(e) => this.combobox.onInputKeydown(e)}
+        type="text"
+        class={'search-input flex-grow outline-none focus:outline-none mx-2'}
+        placeholder={this.props.placeholder}
+        value={this.props.controller.state.facetSearchQuery}
       />
     );
   }
 
-  private get facetSearchResultList() {
-    return this.facetSearchResults.map((searchResult) => (
-      <div onClick={() => this.resultSelected.emit(searchResult)}>
-        {searchResult.displayValue} {searchResult.count}
-      </div>
-    ));
+  private get suggestions() {
+    return (this.props.controller.facetSearchState
+      .values as BaseFacetSearchResult[]).map((suggestion, index) => {
+      return (
+        <li
+          onClick={() =>
+            this.props.controller.facetSearch.select(suggestion as any)
+          }
+          onMouseDown={(e) => e.preventDefault()}
+          part="suggestion"
+          class="suggestion cursor-pointer flex flex-row items-center px-2 text-sm"
+          value={index}
+        >
+          <span class="label font-semibold whitespace-nowrap overflow-ellipsis overflow-hidden">
+            {suggestion.rawValue}
+          </span>
+          <span class="number-of-values ml-auto text-on-background-variant">
+            {suggestion.count}
+          </span>
+        </li>
+      );
+    });
   }
 
   private get showMoreSearchResults() {
-    if (!this.moreValuesAvailable) {
+    if (this.props.controller.facetSearchState.values.length === 0) {
       return null;
     }
 
     return (
-      <button onClick={() => this.showMoreResults.emit()}>show more</button>
+      <button
+        class="px-1 text-primary"
+        onClick={() => this.props.controller.state.facet.showMoreValues()}
+      >
+        show more
+      </button>
     );
   }
 
   public render() {
     return (
-      <div>
-        {this.facetSearchInput}
-        {this.facetSearchResultList}
-        {this.showMoreSearchResults}
+      <div class="search-box relative">
```

Comment:
Similar comment as above regarding splitting up in multiple logical blocks, or extracting the classes generation code in getter function.